extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -79,6 +80,7 @@ protected enum State
   private final TaskLogs taskLogs;
   private final KubernetesPeonClient kubernetesClient;
   private final ObjectMapper mapper;
+  private final BiConsumer<State, String> stateListener;
```

Comment:
Might be nicer to create a new state listener interface instead of using a `BiConsumer`.

Reply:
Added.